code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+	local_bs_list -= src
+	if(local_bs_list.len == 0)
+		return ..()
+
+	if(local_bs_list.len == 1)
+		do_teleport(user ,local_bs_list[1])
+		return
+
+	in_use = TRUE
+
+	var/list/assoc_list = list()
+
+	for(var/BSC in local_bs_list)
+		var/area/bsc_area = get_area(BSC)
+		var/name = "[bsc_area.name] bluespace slimic pylon"
+		var/counter = 0
+
+		do
+			counter++
+		while(assoc_list["[name]([counter])"])
+
+		name += "([counter])"
+
+		assoc_list[name] = BSC
+
+	var/chosen_input = input(user,"What destination do you want to choose",null) as null|anything in assoc_list
+	in_use = FALSE
+
+	if(!chosen_input)
+		return
+
+	do_teleport(user ,assoc_list[chosen_input])
+
+/obj/structure/slime_crystal/sepia
+	colour = "sepia"
+
+/obj/structure/slime_crystal/sepia/on_mob_enter(mob/living/affected_mob)
+	. = ..()
+	ADD_TRAIT(affected_mob,TRAIT_NOBREATH,type)
+	ADD_TRAIT(affected_mob,TRAIT_NOCRITDAMAGE,type)
+	ADD_TRAIT(affected_mob,TRAIT_RESISTLOWPRESSURE,type)
+	ADD_TRAIT(affected_mob,TRAIT_RESISTHIGHPRESSURE,type)
+	ADD_TRAIT(affected_mob,TRAIT_NOSOFTCRIT,type)
+	ADD_TRAIT(affected_mob,TRAIT_NOHARDCRIT,type)
+
+/obj/structure/slime_crystal/sepia/on_mob_leave(mob/living/affected_mob)
+	. = ..()
+	REMOVE_TRAIT(affected_mob,TRAIT_NOBREATH,type)
+	REMOVE_TRAIT(affected_mob,TRAIT_NOCRITDAMAGE,type)
+	REMOVE_TRAIT(affected_mob,TRAIT_RESISTLOWPRESSURE,type)
+	REMOVE_TRAIT(affected_mob,TRAIT_RESISTHIGHPRESSURE,type)
+	REMOVE_TRAIT(affected_mob,TRAIT_NOSOFTCRIT,type)
+	REMOVE_TRAIT(affected_mob,TRAIT_NOHARDCRIT,type)
+
+/obj/structure/cerulean_slime_crystal
+	name = "Cerulean slime poly-crystal"
+	desc = "Translucent and irregular, it can duplicate matter on a whim"
+	anchored = TRUE
+	density = FALSE
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "cerulean_crystal"
+	max_integrity = 5
+	var/stage = 0
+	var/max_stage = 5
+
+/obj/structure/cerulean_slime_crystal/Initialize()
+	. = ..()
+	transform *= 1/(max_stage-1)
+	stage_growth()
+
+/obj/structure/cerulean_slime_crystal/proc/stage_growth()
+	if(stage == max_stage)
+		return
+
+	stage++
+
+	if(stage == 4)
+		density = TRUE
+
+	var/matrix/M = new
+	M.Scale(1/(max_stage-1) * stage)
+
+	animate(src,transform = matrix(), time = 60 SECONDS)
+
+	addtimer(CALLBACK(src,.proc/stage_growth),60 SECONDS)
+
+/obj/structure/cerulean_slime_crystal/Destroy()
+	var/obj/item/cerulean_slime_crystal/crystal = new /obj/item/cerulean_slime_crystal(get_turf(src))
+	crystal.amt = stage
+	return ..()
+
+/obj/structure/slime_crystal/cerulean
+	colour = "cerulean"
+
+/obj/structure/slime_crystal/cerulean/process()
+	for(var/turf/T in orange(2,src))
+		var/obj/structure/cerulean_slime_crystal/CSC = locate() in range(1,T)
+		if(CSC)
+			continue
+		new /obj/structure/cerulean_slime_crystal(T)
+
+/obj/structure/slime_crystal/pyrite
+	colour = "pyrite"
+
+/obj/structure/slime_crystal/pyrite/Initialize()
+	. = ..()
+	change_colour()
+
+/obj/structure/slime_crystal/pyrite/process()
+	return PROCESS_KILL
```

Comment:
Again, there should be a var on the slime_crystal base that lets you not start process in the first place